File: go/fiber/internal/modules/account/dto/find_all_query_account_dto.go
package dto

import "tcc/internal/common/helper"

type FindAllQueryAccountDto struct {
	CPF string `validate:"required,cpf"`
}

func (dto *FindAllQueryAccountDto) Transform() {
	// CPF
	dto.CPF = helper.CleanDigits(dto.CPF)
}

File: rust/axum/src/modules/account/dto/find_all_query_account_dto.rs
use serde::{Deserialize, Serialize};
use validator::Validate;
use crate::common::validator::validate_cpf;
use crate::common::helper::clean_digits::clean_digits;

#[derive(Debug, Serialize, Deserialize, Validate)]
pub struct FindAllQueryAccountDto {
    #[validate(custom(function = "validate_cpf"))]
    pub cpf: String,
}

impl FindAllQueryAccountDto {
    pub fn transform(&mut self) {
        self.cpf = clean_digits(&self.cpf);
    }
}